File: lib/vger.js
/*
*	Vger bot
*	a simple node bot who can learn commands
*	meant to run as a RESTful web service
*	with a simple web based front end to communicate
*	
*	Veejer is meant to be a useful learning tool
*	as well as a potential aid in automation...
*	so far he is none of these :)
*	
*	The interface is inspired by the interface provided
*	for Zakim on the W3C IRC bridge
*/

var fs = require("fs")

/*
*   function ask
*   @parameter q - string
*   @parameter callback - function
*/
exports.ask = function(q, callback){
    // fire off the process of answering the question
    vger.getAnswer(q, function(ans){
        callback(ans)
    })
}

/*
*   function init
*/
exports.init = function(){
    vger.initialize()
}

// define the main functions for vger here
var vger = {
    initialize : function(){
        // initialize vger by retreiving the stored commands
        console.log("initializing Vger")
        
    },
    getAnswer : function(q, callback) {
        // process the answer out to get actor/verb/object
        callback("Why did you say "+q+"?")
    }
}

File: lib/vger.py
class Vger :
	# initialize Vger
	def _init(self) :
		print "Intializing Vger"
		print "I am Vger. What can I do for you?"

	# ask function to process questions
	def ask(self, q) :
		if q == "quit":
			print "quitting"
		elif q == "how are you?":
			print "I am fine, thank you."
			self.askNew()
		else:
			print "I don't understand your command, %s" % q
			self.askNew()

	# ask a new question
	def askNew(self) :
		qnew = raw_input(prompt)
		self.ask(qnew)


vger = Vger()
prompt = ">>"

# run the init on Vger
vger._init()
# ask the first question to start the looping
q = raw_input(prompt)
vger.ask(q)

